mycroft/skills/phillips_hue/__init__.py

```diff
@@ -0,0 +1,283 @@
+            if self.ip == '':
+                self.ip = _discover_bridge()
+            self.bridge = Bridge(self.ip, self.username)
+            self.username = self.bridge.username
+        except DeviceNotFoundException:
+            self.speak_dialog('bridge.not.found')
+            return False
+        except PhueRegistrationException:
+            self.speak_dialog('connect.to.bridge')
+            i = 0
+            while i < 30:
+                time.sleep(1)
+                try:
+                    self.bridge = Bridge(self.ip)
+                except PhueRegistrationException:
+                    continue
+                else:
+                    break
+            if not self.connected:
+                self.speak_dialog('failed.to.register')
+                return False
+            else:
+                self.speak_dialog('successfully.registered')
+        self.all_lights = Group(self.bridge, 0)
+        return True
+
+    def initialize(self):
+        self.load_data_files(dirname(__file__))
+
+        turn_off_intent = IntentBuilder("TurnOffIntent")\
+            .require("TurnOffKeyword").build()
+        self.register_intent(turn_off_intent, self.handle_intent)
+
+        turn_on_intent = IntentBuilder("TurnOnIntent")\
+            .require("TurnOnKeyword").build()
+        self.register_intent(turn_on_intent, self.handle_intent)
+
+        activate_scene_intent = IntentBuilder("ActivateSceneIntent")\
+            .require("ActivateSceneKeyword").build()
+        self.register_intent(activate_scene_intent,
+                             self.handle_intent)
+
+        decrease_brightness_intent = IntentBuilder("DecreaseBrightnessIntent")\
+            .require("DecreaseBrightnessKeyword").build()
+        self.register_intent(decrease_brightness_intent,
+                             self.handle_intent)
+
+        increase_brightness_intent = IntentBuilder("IncreaseBrightnessIntent")\
+            .require("IncreaseBrightnessKeyword").build()
+        self.register_intent(increase_brightness_intent,
+                             self.handle_intent)
+
+        decrease_color_temperature_intent =\
+            IntentBuilder("DecreaseColorTemperatureIntent")\
+            .require("DecreaseColorTemperatureKeyword").build()
+        self.register_intent(decrease_color_temperature_intent,
+                             self.handle_intent)
+
+        increase_color_temperature_intent =\
+            IntentBuilder("IncreaseColorTemperatureIntent")\
+            .require("IncreaseColorTemperatureKeyword").build()
+        self.register_intent(increase_color_temperature_intent,
+                             self.handle_intent)
+
+        connect_lights_intent = \
+            IntentBuilder("ConnectLightsIntent") \
+            .require("ConnectLightsKeyword").build()
+        self.register_intent(connect_lights_intent,
+                             self.handle_intent)
+
+    def handle_intent(self, message):
```

Comment:
I'm not so concerned with the odds, as with consequences. If we don't have error checking like this, and something goes wrong, the user will hear a generic error about something going wrong with the skill. This error is entirely unhelpful, as it doesn't provide any information on how to resolve the error (e.g. try again, fix your config, re-connect). We also can't do things such as attempt to re-connect automatically, if we get a 'no route to host' exception (as I mentioned I should be doing, in one of my comments yesterday).

Also, I don't think the connection stuff should happen in `__init__`. In general, I don't think it's a good idea to have long running operations (especially network operations) happen in `__init__`. Here, I think it would be particularly bad, because the bridge discovery takes up to 5 seconds, which means for anyone without a hue bridge, their boot time would increase by 5 seconds for nothing. Also, the initial connection requires interaction from the user to register the application with the bridge, which would be strange to handle during bootup.

I didn't see documentation for the `initialize` function, but it looks like it is called in `skills.core.load_skill`, shortly after `__init__`, meaning it would have the same issues.

I'm also not quite sure what you mean by 'fail.' Do you mean raise an exception, so the skill isn't loaded? Speak a dialog to the user indicating the failure? Something else?